src/main/java/com/salesforce/storm/spout/dynamic/coordinator/SpoutRunner.java

```diff
@@ -121,21 +121,17 @@ public void run() {
             long lastFlush = getClock().millis();
 
             // Loop forever until someone requests the spout to stop
-            while (!isStopRequested() && !spout.isStopRequested() && !Thread.interrupted()) {
+            while (!isStopRequested() && !spout.isCompleted() && !Thread.interrupted()) {
                 // First look for any new tuples to be emitted.
                 final Message message = spout.nextTuple();
                 if (message != null) {
                     try {
                         getVirtualSpoutMessageBus().publishMessage(message);
                     } catch (final InterruptedException interruptedException) {
                         logger.error("Shutting down due to interruption {}", interruptedException.getMessage(), interruptedException);
```

Comment:
I think you need to set Thread.interupted here and call continue to end the loop.  Or call break.   Basically someway to end the loop.  I could be wrong, but I believe when you catch the exception, the interrupted flag gets cleared.

Reply:
Not sure I follow, what do you mean @Crim ?

Reply:
yea

Reply:
https://stackoverflow.com/questions/10401947/methods-that-clear-the-thread-interrupt-flag